### ISSUE TITLE
Waydroid crash when switch back to its workspace

### ISSUE THREAD

**Issue by zayn7lie** [hyprwm/Hyprland#6923]
### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
$ hyprctl systeminfo
Hyprland, built from branch main at commit d7ea1b7785f31e9fd6477490309721a19a835592  (xwayland: break cyclic loop of parents (6722)).
Date: Mon Jul 1 06:26:08 2024
Tag: v0.41.2-14-gd7ea1b77, commits: 4900

flags: (if any)


System Information:
System name: Linux
Node name: zayn7lie-acl
Release: 6.9.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Fri, 12 Jul 2024 00:06:53 +0000


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-LP GT2 [Iris Xe Graphics] [8086:9a49] (rev 03) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

```
</details>

### Description

After launching waydroid, and moving the focus to another workspaces for a time, waydroid might crash and restart after moving the focus back to it. 

After restarting by itself, it might lose the network connection.

I'm not sure whether it is the problem of Hyprland or Waydroid. Could someone help me to try under other WMs?

### How to reproduce

Open waydroid and leave it for a time, then move the focus back to it.

### Crash reports, logs, images, videos

```sh
### while switching back to the workspace of waydroid
[Tue Jul 23 09:47:16 2024] binder_linux: release 14082:14082 transaction 2747439 in, still active
[Tue Jul 23 09:47:16 2024] binder_linux: send failed reply for transaction 2747439 to 14103:14103
[Tue Jul 23 09:47:16 2024] init: Service 'vendor.hwcomposer-2-1' (pid 65) received signal 6
[Tue Jul 23 09:47:16 2024] init: Sending signal 9 to service 'vendor.hwcomposer-2-1' (pid 65) process group...
[Tue Jul 23 09:47:16 2024] libprocessgroup: Successfully killed process cgroup uid 1000 pid 65 in 0ms
[Tue Jul 23 09:47:16 2024] init: Sending signal 9 to service 'surfaceflinger' (pid 86) process group...
[Tue Jul 23 09:47:16 2024] libprocessgroup: Successfully killed process cgroup uid 1000 pid 86 in 0ms
[Tue Jul 23 09:47:16 2024] init: starting service 'vendor.hwcomposer-2-1'...
[Tue Jul 23 09:47:16 2024] libprocessgroup: Failed to make and chown /acct/uid_1000: Read-only file system
[Tue Jul 23 09:47:16 2024] init: createProcessGroup(1000, 3494) failed for service 'vendor.hwcomposer-2-1': Read-only file system
[Tue Jul 23 09:47:17 2024] binder_linux: release 14103:14140 transaction 2747523 in, still active
[Tue Jul 23 09:47:17 2024] binder_linux: send failed reply for transaction 2747523 to 17268:17407
[Tue Jul 23 09:47:17 2024] init: Service 'surfaceflinger' (pid 86) received signal 6
[Tue Jul 23 09:47:17 2024] init: Sending signal 9 to service 'surfaceflinger' (pid 86) process group...
[Tue Jul 23 09:47:17 2024] libprocessgroup: Successfully killed process cgroup uid 1000 pid 86 in 0ms
[Tue Jul 23 09:47:17 2024] init: Sending signal 9 to service 'zygote' (pid 58) process group...
[Tue Jul 23 09:47:17 2024] libprocessgroup: Successfully killed process cgroup uid 0 pid 58 in 0ms
[Tue Jul 23 09:47:17 2024] init: starting service 'surfaceflinger'...
[Tue Jul 23 09:47:17 2024] init: Could not create socket 'pdx/system/vr/display/client': Failed to bind socket 'pdx/system/vr/display/client': No such file or directory
[Tue Jul 23 09:47:17 2024] init: Could not create socket 'pdx/system/vr/display/manager': Failed to bind socket 'pdx/system/vr/display/manager': No such file or directory
[Tue Jul 23 09:47:17 2024] init: Could not create socket 'pdx/system/vr/display/vsync': Failed to bind socket 'pdx/system/vr/display/vsync': No such file or directory
[Tue Jul 23 09:47:17 2024] libprocessgroup: Failed to make and chown /acct/uid_1000: Read-only file system
[Tue Jul 23 09:47:17 2024] init: createProcessGroup(1000, 3507) failed for service 'surfaceflinger': Read-only file system
[Tue Jul 23 09:47:17 2024] libprocessgroup: Failed to open /dev/stune/foreground/tasks: No such file or directory: No such file or directory
[Tue Jul 23 09:47:17 2024] libprocessgroup: Failed to apply HighPerformance task profile: No such file or directory
[Tue Jul 23 09:47:17 2024] init: starting service 'bootanim'...
[Tue Jul 23 09:47:17 2024] libprocessgroup: Failed to make and chown /acct/uid_1003: Read-only file system
[Tue Jul 23 09:47:17 2024] init: createProcessGroup(1003, 3527) failed for service 'bootanim': Read-only file system
[Tue Jul 23 09:47:17 2024] init: Control message: Processed ctl.start for 'bootanim' from pid: 3507 (/system/bin/surfaceflinger)
[Tue Jul 23 09:47:17 2024] libprocessgroup: Failed to open /dev/stune/top-app/tasks: No such file or directory: No such file or directory
[Tue Jul 23 09:47:17 2024] libprocessgroup: Failed to apply MaxPerformance task profile: No such file or directory
[Tue Jul 23 09:47:18 2024] binder_linux: 17268:17268 transaction failed 29189/-22, size 88-0 line 2989
[Tue Jul 23 09:47:18 2024] binder_linux: 14273:14307 transaction failed 29189/-22, size 88-0 line 2989
### crash here
[Tue Jul 23 09:47:33 2024] binder_linux: 15913:15913 transaction failed 29189/-22, size 88-0 line 2989
[Tue Jul 23 09:47:38 2024] binder_linux: 14273:14293 transaction failed 29189/-22, size 88-0 line 2989
[Tue Jul 23 09:47:40 2024] binder_linux: 14273:14294 transaction failed 29189/-22, size 88-0 line 2989
[Tue Jul 23 09:47:40 2024] binder_linux: 14273:14291 transaction failed 29189/-22, size 200-8 line 2989
[Tue Jul 23 09:47:40 2024] binder_linux: 14273: binder_alloc_buf, no vma
[Tue Jul 23 09:47:40 2024] binder_linux: 17268:17460 transaction failed 29189/-3, size 148-0 line 3166
[Tue Jul 23 09:47:40 2024] binder_linux: send failed reply for transaction 2774854 to 17268:17327
[Tue Jul 23 09:47:40 2024] binder_linux: 14098:14247 transaction failed 29189/-22, size 124-8 line 2989
[Tue Jul 23 09:47:40 2024] init: Service 'zygote' (pid 58) received signal 9
[Tue Jul 23 09:47:40 2024] init: Sending signal 9 to service 'zygote' (pid 58) process group...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Successfully killed process cgroup uid 0 pid 58 in 0ms
[Tue Jul 23 09:47:40 2024] init: starting service 'exec 20 (/system/bin/vdc volume abort_fuse)'...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Failed to make and chown /acct/uid_1000: Read-only file system
[Tue Jul 23 09:47:40 2024] init: createProcessGroup(1000, 3582) failed for service 'exec 20 (/system/bin/vdc volume abort_fuse)': Read-only file system
[Tue Jul 23 09:47:40 2024] init: Command 'write /sys/power/state on' action=onrestart (<Service 'zygote' onrestart>:2) took 0ms and failed: Unable to write to file '/sys/power/state': open() failed: Read-only file system
[Tue Jul 23 09:47:40 2024] init: Sending signal 9 to service 'audioserver' (pid 81) process group...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Successfully killed process cgroup uid 1041 pid 81 in 0ms
[Tue Jul 23 09:47:40 2024] init: Sending signal 9 to service 'cameraserver' (pid 152) process group...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Successfully killed process cgroup uid 1047 pid 152 in 0ms
[Tue Jul 23 09:47:40 2024] init: Sending signal 9 to service 'media' (pid 164) process group...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Successfully killed process cgroup uid 1013 pid 164 in 0ms
[Tue Jul 23 09:47:40 2024] init: Sending signal 9 to service 'netd' (pid 57) process group...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Successfully killed process cgroup uid 0 pid 57 in 0ms
[Tue Jul 23 09:47:40 2024] init: Sending signal 9 to service 'wificond' (pid 166) process group...
[Tue Jul 23 09:47:40 2024] libprocessgroup: Successfully killed process cgroup uid 1010 pid 166 in 0ms
[Tue Jul 23 09:47:40 2024] init: processing action (init.svc.media=*) from (/system/etc/init/mediaserver.rc:1)
[Tue Jul 23 09:47:40 2024] init: starting service 'zygote'...
[Tue Jul 23 09:47:40 2024] init: Created socket '/dev/socket/zygote', mode 660, user 0, group 1000
[Tue Jul 23 09:47:40 2024] init: Created socket '/dev/socket/usap_pool_primary', mode 660, user 0, group 1000
[Tue Jul 23 09:47:40 2024] libprocessgroup: Failed to make and chown /acct/uid_0: Read-only file system
[Tue Jul 23 09:47:40 2024] init: createProcessGroup(0, 3583) failed for service 'zygote': Read-only file system
[Tue Jul 23 09:47:40 2024] libprocessgroup: Failed to open /dev/stune/top-app/tasks: No such file or directory: No such file or directory
[Tue Jul 23 09:47:40 2024] libprocessgroup: Failed to apply MaxPerformance task profile: No such file or directory
[Tue Jul 23 09:47:41 2024] init: Service 'exec 20 (/system/bin/vdc volume abort_fuse)' (pid 3582) exited with status 0 oneshot service took 0.004000 seconds in background
[Tue Jul 23 09:47:41 2024] init: Sending signal 9 to service 'exec 20 (/system/bin/vdc volume abort_fuse)' (pid 3582) process group...
[Tue Jul 23 09:47:41 2024] libprocessgroup: Successfully killed process cgroup uid 1000 pid 3582 in 0ms
[Tue Jul 23 09:47:41 2024] binder_linux: 14684:14706 transaction failed 29189/-22, size 164-0 line 2989
[Tue Jul 23 09:47:42 2024] init: Control message: Processed ctl.start for 'idmap2d' from pid: 3590 (system_server)
[Tue Jul 23 09:47:42 2024] binder_linux: 14171:14472 transaction failed 29189/-22, size 188-0 line 2989
[Tue Jul 23 09:47:42 2024] init: processing action (sys.sysctl.extra_free_kbytes=*) from (/system/etc/init/hw/init.rc:995)
[Tue Jul 23 09:47:43 2024] binder_linux: 14628:14764 transaction failed 29189/-22, size 164-0 line 2989
[Tue Jul 23 09:47:43 2024] binder_linux: 14628:14859 transaction failed 29189/-22, size 84-0 line 2989
[Tue Jul 23 09:47:43 2024] binder_linux: 15913:15913 transaction failed 29189/-22, size 148-8 line 2989
[Tue Jul 23 09:47:43 2024] binder_linux: 14937:15203 transaction failed 29189/-22, size 280-8 line 2989
[Tue Jul 23 09:47:43 2024] init: Untracked pid 897 received signal 9
[Tue Jul 23 09:47:43 2024] init: Untracked pid 605 received signal 9
### crash again...
[Tue Jul 23 09:47:48 2024] binder_transaction: 3 callbacks suppressed
[Tue Jul 23 09:47:48 2024] binder_linux: 15108:15108 transaction failed 29189/-22, size 104-8 line 2989
[Tue Jul 23 09:47:48 2024] init: Untracked pid 1064 received signal 9
[Tue Jul 23 09:47:49 2024] init: Service 'bootanim' (pid 3527) exited with status 0 oneshot service took 32.495998 seconds in background
[Tue Jul 23 09:47:49 2024] init: Sending signal 9 to service 'bootanim' (pid 3527) process group...
[Tue Jul 23 09:47:49 2024] libprocessgroup: Successfully killed process cgroup uid 1003 pid 3527 in 0ms
[Tue Jul 23 09:47:49 2024] init: processing action (sys.boot_completed=1) from (/system/etc/init/hw/init.rc:986)
[Tue Jul 23 09:47:49 2024] init: starting service 'exec 21 (/bin/rm -rf /data/per_boot)'...
[Tue Jul 23 09:47:49 2024] libprocessgroup: Failed to make and chown /acct/uid_1000: Read-only file system
[Tue Jul 23 09:47:49 2024] init: createProcessGroup(1000, 4066) failed for service 'exec 21 (/bin/rm -rf /data/per_boot)': Read-only file system
[Tue Jul 23 09:47:49 2024] init: SVC_EXEC service 'exec 21 (/bin/rm -rf /data/per_boot)' pid 4066 (uid 1000 gid 1000+0 context default) started; waiting...
[Tue Jul 23 09:47:49 2024] init: Service 'exec 21 (/bin/rm -rf /data/per_boot)' (pid 4066) exited with status 0 waiting took 0.006000 seconds
[Tue Jul 23 09:47:49 2024] init: Sending signal 9 to service 'exec 21 (/bin/rm -rf /data/per_boot)' (pid 4066) process group...
[Tue Jul 23 09:47:49 2024] libprocessgroup: Successfully killed process cgroup uid 1000 pid 4066 in 0ms
[Tue Jul 23 09:47:49 2024] init: processing action (sys.boot_completed=1) from (/system/etc/init/flags_health_check.rc:7)
[Tue Jul 23 09:47:49 2024] init: processing action (sys.boot_completed=1) from (/system/etc/init/init.waydroid.rc:6)
[Tue Jul 23 09:47:49 2024] init: processing action (sys.boot_completed=1) from (/system/etc/init/logd.rc:33)
[Tue Jul 23 09:47:49 2024] init: starting service 'logd-auditctl'...
[Tue Jul 23 09:47:49 2024] libprocessgroup: Failed to make and chown /acct/uid_1036: Read-only file system
[Tue Jul 23 09:47:49 2024] init: createProcessGroup(1036, 4123) failed for service 'logd-auditctl': Read-only file system
[Tue Jul 23 09:47:49 2024] init: processing action (sys.user.0.ce_available=true) from (/system/etc/init/wifi.rc:21)
[Tue Jul 23 09:47:49 2024] init: Service 'logd-auditctl' (pid 4123) exited with status 0 oneshot service took 0.003000 seconds in background
[Tue Jul 23 09:47:49 2024] init: Sending signal 9 to service 'logd-auditctl' (pid 4123) process group...
[Tue Jul 23 09:47:49 2024] libprocessgroup: Successfully killed process cgroup uid 1036 pid 4123 in 0ms
[Tue Jul 23 09:47:49 2024] init: Untracked pid 3141 received signal 2
[Tue Jul 23 09:47:54 2024] init: Sending signal 9 to service 'idmap2d' (pid 154) process group...
[Tue Jul 23 09:47:54 2024] libprocessgroup: Successfully killed process cgroup uid 1000 pid 154 in 0ms
[Tue Jul 23 09:47:54 2024] init: Control message: Processed ctl.stop for 'idmap2d' from pid: 3590 (system_server)
[Tue Jul 23 09:47:54 2024] binder_linux: 14007 RLIMIT_NICE not set
[Tue Jul 23 09:47:54 2024] binder_linux: 14007 RLIMIT_NICE not set
[Tue Jul 23 09:48:11 2024] binder_linux: 15596:15645 transaction failed 29189/-22, size 84-0 line 2989
[Tue Jul 23 09:48:11 2024] binder_linux: 15596:15645 transaction failed 29189/-22, size 116-0 line 2989
[Tue Jul 23 09:48:11 2024] binder_linux: 15596:15645 transaction failed 29189/-22, size 1908-8 line 2989
[Tue Jul 23 09:48:11 2024] init: Untracked pid 1552 received signal 9
[Tue Jul 23 09:48:32 2024] binder_linux: 14007 RLIMIT_NICE not set
[Tue Jul 23 09:48:32 2024] binder_linux: 14007 RLIMIT_NICE not set
[Tue Jul 23 09:48:32 2024] binder_linux: 14007 RLIMIT_NICE not set
[Tue Jul 23 09:48:32 2024] binder_linux: 19112: binder_alloc_buf size 1056768 failed, no address space
[Tue Jul 23 09:48:32 2024] binder_linux: allocated: 648 (num: 12 largest: 192), free: 1039736 (num: 7 largest: 1033040)
[Tue Jul 23 09:48:32 2024] binder_linux: 19168:19631 transaction failed 29201/-28, size 1056768-0 line 3166
### no network connection
```

**Comment by zayn7lie** [comment]
Partially fixed with #7582 , but other actions like sleep, full screen of other windows, etc. would still have the same issue.

**Comment by zayn7lie** [comment]
Since it seems to not be a Hyprland-specific problem, this issue closed. More info: [waydroid/waydroid#1486](https://github.com/waydroid/waydroid/issues/1486)
